src/react/components/common/assetPreview/videoAsset.tsx

```diff
@@ -225,4 +288,89 @@ export class VideoAsset extends React.Component<IVideoAssetProps> {
         }
     }
+
+    /**
+     * @name - Ensure seek is on valid keyframe
+     * @description - Move to the nearest key frame from where the video's current
+     * position is
+     * @returns true if moved to a new position; false otherwise
+     */
+    private ensureSeekIsOnValidKeyframe(): boolean {
+        if (!this.props.additionalSettings) {
+            return false;
+        }
+        const keyFrameTime = (1 / this.props.additionalSettings.videoSettings.frameExtractionRate);
+        const timestamp = this.getCurrentVideoPlayerState().currentTime;
+
+        // Calculate the nearest key frame
+        const numberKeyFrames = Math.round(timestamp / keyFrameTime);
+        const seekTime = (numberKeyFrames * keyFrameTime);
+        if (seekTime !== timestamp) {
+            this.seekToTime(seekTime);
+        }
+
+        return seekTime !== timestamp;
+    }
+
+    /**
+     * @name - Add Asset Timeline Tags
+     * @description - Draws small lines to show where visited and tagged frames are on
+     * the video line
+     * @member childAssets - Array of child assets in the video
+     * @member videoDuration - Length (in seconds) of the video
+     */
+    private addAssetTimelineTags(childAssets: any[], videoDuration: number) {
+        let progressHolderElement: Element = null;
+        const assetTimelineTagLines = this.getRenderedAssetTagLinesElements(childAssets, videoDuration);
+        if (!this.timelineElement) {
+            const editorElement = document.querySelector("div.editor-page-content-body");
+            if (editorElement) {
+                progressHolderElement = editorElement.getElementsByClassName("video-react-progress-control")[0];
+            }
+            // If we found an element to hold the tags, add them to it
+            if (progressHolderElement) {
+                this.timelineElement = document.createElement("div");
+                progressHolderElement.appendChild(this.timelineElement);
+            }
+        }
+
+        // Render the child asset elmements to the dom
+        if (this.timelineElement) {
+            ReactDOM.render(assetTimelineTagLines, this.timelineElement);
+        }
+    }
+
+    /**
+     * @name - Get Rendered Asset Tag Lines Elements
+     * @description - Gets the elements for the rendered asset tag lines
+     * @member childAssets - Array of child assets in the video
+     * @member videoDuration - Length (in seconds) of the video
+     */
+    private getRenderedAssetTagLinesElements(childAssets: any[], videoDuration: number) {
+        const tagTimeLines: any = [];
+
+        // Add some markers for frames that have been visited with yellow and tagged with green
+        for (const childAsset of childAssets) {
+            // Calcualte the left position
+            const childPosition: number = (childAsset.timestamp / videoDuration);
+            tagTimeLines.push(<div key={childAsset.timestamp}
+                onClick = {this.handleAssetTimelineClick}
```

Comment:
FYI, you can also specify an inline func to pass params ex. `onClick={() => this.handleAssetTimelineClick(childAsset.timestamp)}` which will retain the correct timestamp in the closure.